### PR TITLE
Add utilities for building `UnitInput` from `CsgTree`

### PR DESCRIPTION
**Pull request by sethrj**
This adds several utility functions necessary to complete the construction of a nice clean happy UnitInput.
- Implemented `find_sorted` for treating a sorted vector like a map
- The postfix builder can remap surface IDs to remove references to unused surfaces
- The postfix builder now returns a {face, logic with face indices} pair
- Renamed `local_orange_outside_volume` to `orange_exterior_volume` and added `orange_global_universe` for UniverseId{0}
- Added a `ProtoMap` function for ordering proto-universes and mapping pointers to `UniverseId`s
- Added a simple function to return the external bounding box from a bounding zone
- Added a utility to determine whether a volume may have internal surface crossings

### CODE REVIEW

File: src/corecel/math/Algorithms.hh [hunk 1]
```diff
@@ -253,6 +253,35 @@ CELER_FORCEINLINE_FUNCTION ForwardIt upper_bound(ForwardIt first,
...
+//---------------------------------------------------------------------------//
+/*!
+ * Find the given element in a sorted range.
+ */
+template<class ForwardIt, class T, class Compare>
+inline CELER_FUNCTION ForwardIt
+find_sorted(ForwardIt first, ForwardIt last, T const& value, Compare comp)
+{
+    auto iter = ::celeritas::lower_bound(first, last, value, comp);
+    if (iter == last || comp(*iter, value) || comp(value, *iter))
```
Comment (elliottbiondo) [inline src/corecel/math/Algorithms.hh:265]: I believe `iter == last` isn't necessary here

File: src/orange/orangeinp/detail/InternalSurfaceFlagger.cc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        CELER_ASSUME(!node.valueless_by_exception());
+        status = static_cast<Status>(std::visit(*this, node));
+    }
+    CELER_ENSURE(status >= 0);
+    return status;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Aliased nodes forward to the alias.
```
Comment (elliottbiondo) [inline src/orange/orangeinp/detail/InternalSurfaceFlagger.cc]: "Forward to aliased node" for sentence structure consistency

**Review by elliottbiondo — APPROVED** [review]

File: src/corecel/math/Algorithms.hh (hunk 1 above)
Comment (sethrj) [reply src/corecel/math/Algorithms.hh:265]: lower_bound returns `last` if the searched-for value is higher than the final value in the sequence. Not testing this means segfaulting on the following `comp` command.

Comment (elliottbiondo) [reply src/corecel/math/Algorithms.hh:265]: When `iter == last`, you can enter the `if` statement and return last, or NOT enter the `if` statement and return `iter`, both returning equivalent results.

Comment (elliottbiondo) [reply src/corecel/math/Algorithms.hh:265]: @sethrj  i.e. this is equivalent to what you have:
```
    if (comp(*iter, value) || comp(value, *iter))
    {
        // Insertion point is off the end, or value is not equal
        return last;
    }
    return iter;
```

Comment (elliottbiondo) [reply src/corecel/math/Algorithms.hh:265]: Ah okay that makes sense

**Pull request merged**